.github/workflows/sycl-post-commit.yml

```diff
@@ -35,7 +35,7 @@ jobs:
       build_cache_root: "/__w/llvm"
       build_cache_suffix: default
       build_artifact_suffix: default
-      build_configure_extra_args: --no-assertions --hip --cuda --native_cpu --cmake-opt="-DSYCL_ENABLE_STACK_PRINTING=ON" --cmake-opt="-DSYCL_LIB_WITH_DEBUG_SYMBOL=ON"      
+      build_configure_extra_args: --no-assertions --hip --cuda --native_cpu --cmake-opt="-DSYCL_ENABLE_STACK_PRINTING=ON" --cmake-opt="-DSYCL_LIB_WITH_DEBUG_SYMBOL=ON --cmake-opt=-DCMAKE_CXX_FLAGS="-Wno-mismatched-new-delete"
```

Comment:
Why was `-Wno-mismatched-new-delete` needed? Can we add a comment here highlighting the need for this flag?

Reply:
Yeah I tried to fix the warning actually but wasn't sure what to do. I think you might have worked in the code the warning is from before so if you have an idea let me know, if not I'll just add a comment. Here's the warning:

```
In file included from /usr/include/c++/13/bits/std_thread.h:43,
                 from /usr/include/c++/13/thread:45,
                 from /tmp/llvm/build/include/sycl/detail/spinlock.hpp:14,
                 from /tmp/llvm/sycl/source/detail/global_handler.hpp:11,
                 from /tmp/llvm/sycl/source/detail/config.hpp:11,
                 from /tmp/llvm/sycl/source/detail/adapter.hpp:11,
                 from /tmp/llvm/sycl/source/detail/ur_utils.hpp:11,
                 from /tmp/llvm/sycl/source/detail/device_binary_image.hpp:10,
                 from /tmp/llvm/sycl/source/detail/device_binary_image.cpp:9:
In member function 'void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = char]',
    inlined from 'std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = char; _Dp = std::default_delete<char>]' at /usr/include/c++/13/bits/unique_ptr.h:404:17,
    inlined from 'static std::unique_ptr<char> sycl::_V1::detail::ZSTDCompressor::DecompressBlob(const char*, size_t, size_t&)' at /tmp/llvm/sycl/source/detail/compression.hpp:139:3,
    inlined from 'void sycl::_V1::detail::CompressedRTDeviceBinaryImage::Decompress()' at /tmp/llvm/sycl/source/detail/device_binary_image.cpp:258:54:
/usr/include/c++/13/bits/unique_ptr.h:99:9: error: 'void operator delete(void*, std::size_t)' called on pointer returned from a mismatched allocation function [-Werror=mismatched-new-delete]
   99 |         delete __ptr;
      |         ^~~~~~~~~~~~
In file included from /tmp/llvm/sycl/source/detail/device_binary_image.cpp:13:
In static member function 'static std::unique_ptr<char> sycl::_V1::detail::ZSTDCompressor::DecompressBlob(const char*, size_t, size_t&)',
    inlined from 'void sycl::_V1::detail::CompressedRTDeviceBinaryImage::Decompress()' at /tmp/llvm/sycl/source/detail/device_binary_image.cpp:258:54:
/tmp/llvm/sycl/source/detail/compression.hpp:119:66: note: returned from 'void* operator new [](std::size_t)'
  119 |     auto dstBuffer = std::unique_ptr<char>(new char[dstBufferSize]);
      |                                                                  ^
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.

```

Reply:
I think I know what's going wrong here. I'll open a PR for fix.

Reply:
Awesome, thanks! Please send it to me once you post it so I can test it, I don't think CI here will replicate the issue because I guess only gcc 13 catches it.

Reply:
I would commit changes in SYCL header and SYCL test in a separate PR. They seem to be useful regardless of Ubuntu version used by CI.

Reply:
Done in https://github.com/intel/llvm/pull/16480, thx